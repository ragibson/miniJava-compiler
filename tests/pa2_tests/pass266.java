// PA2 pass AST reflects associativity of nested if stmts
class B {
    public void foo() {
        // nested if stmts should associate in order suggested
        // by indentation

        if (b)
            if (c)
                x = 1;
            else
                x = 2;
        else
            if (d)
                x = 11;
            else
                x = 22;


        if (true)
            if (false)
                x = 3;
            else
                x = 4;

       
        if (!true)
            x = 33;
        else
            if (!false)
                x = 44;
            else
                x = 55;
    }
}
