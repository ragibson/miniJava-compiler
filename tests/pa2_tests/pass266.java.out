********** miniJava Compiler (PA2) **********
Syntactic Analysis ... 
Syntactic Analysis OK
======= AST Display =========================
Package
  ClassDeclList [1]
  . ClassDecl
  .   "B" classname
  .   FieldDeclList [0]
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "foo" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [3]
  .   .   . IfStmt
  .   .   .   RefExpr
  .   .   .     IdRef
  .   .   .       "b" Identifier
  .   .   .   IfStmt
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "c" Identifier
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "1" IntLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "2" IntLiteral
  .   .   .   IfStmt
  .   .   .     RefExpr
  .   .   .       IdRef
  .   .   .         "d" Identifier
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "11" IntLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "22" IntLiteral
  .   .   . IfStmt
  .   .   .   LiteralExpr
  .   .   .     "true" BooleanLiteral
  .   .   .   IfStmt
  .   .   .     LiteralExpr
  .   .   .       "false" BooleanLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "3" IntLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "4" IntLiteral
  .   .   . IfStmt
  .   .   .   UnaryExpr
  .   .   .     "!" Operator
  .   .   .       LiteralExpr
  .   .   .         "true" BooleanLiteral
  .   .   .   AssignStmt
  .   .   .     IdRef
  .   .   .       "x" Identifier
  .   .   .     LiteralExpr
  .   .   .       "33" IntLiteral
  .   .   .   IfStmt
  .   .   .     UnaryExpr
  .   .   .       "!" Operator
  .   .   .         LiteralExpr
  .   .   .           "false" BooleanLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "44" IntLiteral
  .   .   .     AssignStmt
  .   .   .       IdRef
  .   .   .         "x" Identifier
  .   .   .       LiteralExpr
  .   .   .         "55" IntLiteral
=============================================
